### DISCUSSION TITLE
Get empty results when no file name is mentioned

### DISCUSSION THREAD

**Discussion by lucky3582** [BurntSushi/ripgrep#1964] [General] [answered]
Hi,
I am new to using rg and found this issue while trying it.

When i search for a string without mentioning the file, I assume it will search in all files in the current directory. But it returns empty results whereas if i mention the file name it gives the results.

ex:
rg "Serial Number is"

this returned empty result

with file name 
rg "Serial Number is" DEBUG-INFO.txt
337489:     Serial Number is NNTMRT109W0G

File is present in the directory and both commands are ran from same directory.

Am i searching it wrong or is it a bug?

Thanks

**Comment by BurntSushi** [comment, marked as answer]
Are you aware that ripgrep does automatic filtering by default? It's explained in detail here: https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#automatic-filtering

Have you checked whether `DEBUG-INFO.txt` is being filtered out? You can check by running with `-uuu`. If the results appear for `DEBUG-INFO.txt` with `-uuu`, then it is either being ignored via gitignore, hidden or binary filters. (Likely the gitignore filter in this case is my guess.) If it's still not clear why it's being filtered out, then run with `--debug`. There should be a log message mentioning `DEBUG-INFO.txt` saying why it was filtered out.

**Reply by lucky3582** [reply]
Thank you. I used -uuu looks like the file is binary. I tried with -a option and it worked. thanks for the response.
